src/main/scala/com.snowplowanalytics.snowplow.storage.kinesis/s3/serializers/ISerializer.scala

```diff
@@ -26,5 +35,21 @@ case class SerializationResult(namedStreams: List[NamedStream], results: List[Em
  * Shared interface for all serializers
  */
 trait ISerializer {
-  def serialize(records: List[ EmitterInput ], baseFilename: String): SerializationResult
+  def serialize(records: List[EmitterInput], baseFilename: String): SerializationResult
+
+  def serializeRecord[T](
+    record: RawRecord,
+    serializer: T,
+    serialize: T => Unit
+  ): ValidatedRecord =
+    try {
+      serialize(serializer)
+      record.success
+    } catch {
+      case e: IOException =>
+        val base64Record = new String(Base64.encodeBase64(record), "UTF-8")
+        FailedRecord(List(s"Error writing raw event to output stream: [$e]"), base64Record).failure
+      case NonFatal(e) =>
+        FailedRecord(List(s"Error writing raw event to output stream: [$e]"), "").failure
```

Comment:
I don't think catching oom errors (or any errors for that matter) is a good idea in the first place that's why I changed it.
Since it's thrown back I don't get why we're trying to log it in the first place (I don't even know what will actually happen for oom or others).

However I agree that the log in case of `NonFatal` is missing.